### PR TITLE
Fix for WFLY-18741, Add a github action shared file to build and test WildFly

### PR DESCRIPTION
**Pull request by jfdenise**
Issue: https://issues.redhat.com/browse/WFLY-18741

An example of the Galleon plugins Github Actions: https://github.com/jfdenise/galleon-plugins/blob/main/.github/workflows/ci.yml
An Action run: https://github.com/jfdenise/galleon-plugins/actions/runs/6862670363

Another Action run example from Galleon project this time (just validate that build + -Dts.layers tests are passing): https://github.com/jfdenise/galleon/actions/runs/6864936818

### CODE REVIEW

File: .github/workflows/shared-wildfly-build-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      - name: Build and Test WildFly SNAPSHOT ${{ inputs.build-arguments }}  ${{ inputs.test-arguments }}
+        run: mvn -U -B -ntp clean install ${{ inputs.build-arguments }} ${{ inputs.test-arguments }}
+        shell: bash
+        # We need this due to long file names on Windows: https://github.com/actions/upload-artifact/issues/240
+      - name: Find reports
+        if: failure()
+        run: |
+          {
+            echo 'SUREFIRE_REPORTS<<EOF'
+            find . -path '**/surefire-reports/*.txt'
```
Comment (jamezp) [inline .github/workflows/shared-wildfly-build-and-test.yml:73]: Minor, but we should leave the *.txt off IMO. I've been capturing the `.xml` files too as they can be helpful. I've written a tool that parses them and gives information to easily get data from the failed tests.

File: .github/workflows/shared-wildfly-build-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      - uses: actions/download-artifact@v3
+        with:
+          name: ${{ inputs.maven-repo-name }}
+          path: .
+      - name: Extract Maven Repo
+        shell: bash
+        run: |
+         tar -xzf ${{ inputs.maven-repo-path }} -C ~
+      - name: Build and Test WildFly SNAPSHOT ${{ inputs.build-arguments }}  ${{ inputs.test-arguments }}
+        run: mvn -U -B -ntp clean install ${{ inputs.build-arguments }} ${{ inputs.test-arguments }}
```
Comment (jamezp) [inline .github/workflows/shared-wildfly-build-and-test.yml:65]: Good call on the `-ntp` argument. I should start using that more :)

File: .github/workflows/shared-wildfly-build-and-test.yml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      java-versions:
+        description: "JDK versions array"
+        required: true
+        default: ""
+        type: string
+      os:
+        description: "OS array"
+        required: true
+        default: ""
+        type: string
```
Comment (jamezp) [inline .github/workflows/shared-wildfly-build-and-test.yml:35]: We should remove the default here if they are required.

**Review by jamezp — CHANGES_REQUESTED** [review]
I think overall this makes sense. One small request, can you add a comment on the top that shows how to use this for a reference? Something similar to what is here https://github.com/wildfly/wildfly/blob/dbf1071f017d21dab7ee5c86aa20302f7586c050/.github/workflows/shared-wildfly-build.yml#L13-L40.

We also might want to note that things like `-DallTests` should not be passed as not all tests will work on GitHub Actions.

File: .github/workflows/shared-wildfly-build-and-test.yml (hunk 1 above)
Comment (jfdenise) [reply .github/workflows/shared-wildfly-build-and-test.yml]: Done.

File: .github/workflows/shared-wildfly-build-and-test.yml (hunk 3 above)
Comment (jfdenise) [reply .github/workflows/shared-wildfly-build-and-test.yml]: Done

**Comment by jfdenise** [comment]
@jamezp , thank-you I added usage, notes and followed your suggestions.

**Review by jamezp — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @jfdenise and @jamezp

**Pull request merged**